serpentTools/parsers/results.py

```diff
@@ -1,15 +1,234 @@
 """Parser responsible for reading the ``*res.m`` files"""
+import re
 
-from serpentTools.objects.readers import BaseReader
+from numpy import array, vstack
 
+from serpentTools.settings import rc
+from serpentTools.objects import splitItems, convertVariableName
+from serpentTools.objects.containers import HomogUniv
+from serpentTools.objects.readers import XSReader
 
-class ResultsReader(BaseReader):
+from serpentTools.messages import (info, debug, SerpentToolsException)
+
+
+class ResultsReader(XSReader):
     """
     Parser responsible for reading and working with result files.
 
+     .. note::
+
+    Branching points for unique burnup steps and universes,
+    will be overwritten
+
     Parameters
     ----------
     filePath: str
-        path to the depletion file
+        path to the results file
+
+    Errors
+    ------
+        If the Serpent version cannot be supported
     """
-    pass
+
+    def __init__(self, filePath):
+        XSReader.__init__(self, filePath, 'xs')
+        self.__fileObj = None
+        self._counter = {'meta': 0, 'rslt': 0, 'univ': 0}
+        self._univlist = list()
+        self.metadata = {}   # descriptive execution data (e.g. execution path)
+        self.resdata = {}    # time-dependent quantities (e.g. k-eff)
+        self.univdata = {}  # universes data (e.g. cross-sections)
+        self._mapSerpentVersion()
+
+    def _read(self):
+        """Read through the results file and store requested data."""
+        info('Preparing to read {}'.format(self.filePath))
+        self._preRead()
+        with open(self.filePath, 'r') as fObject:
+            for tline in fObject:
+                self._processResults(tline)
+        info('Done reading results file')
+
+    def _processResults(self, tline):
+        """Performs the main processing of the results."""
+        #  Not an empty line or a comment
+        if tline.strip() and not re.search(r'\%', tline) and \
+                re.search(r'=', tline) and not re.search(r'idx = ', tline):
```

Comment:
Do these have to be regular expressions? We only care if lines
1. Are not blank (treated with `tline.strip()`)
1. Are not comments
1. Do no adjust `idx` which is at the start of the line.

What other cases have you seen that may require these regular expressions?